python/pyiceberg/expressions/literals.py

```diff
@@ -82,15 +76,15 @@ def to(self, type_var: IcebergType) -> Literal:
         ...  # pragma: no cover
 
     def __repr__(self) -> str:
-        return f"{type(self).__name__}({self.value})"
+        return f"{type(self).__name__}({self.value!r})"
```

Comment:
In most places we use `repr(self.value)` rather than `!r`. Is there a best practice to choose between the two?

Reply:
Because mypy knows that it also can be `bytes`, it recommends adding the raw option:
```
➜  python git:(fd-remove-dataclass) ✗ pre-commit run --all-files mypy
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

python/pyiceberg/expressions/literals.py:79: error: On Python 3 formatting "b'abc'" with "{}" produces "b'abc'", not "abc"; use "{!r}" if this is desired behavior  [str-bytes-safe]
```